src/core/src/pass/manager.cpp

```diff
@@ -89,89 +130,184 @@ class stopwatch {
+                           << std::endl;
+                } else {
+                    m_file << "t;" << name << ";" << m_manager_name << ";" << stopwatch.get_timer_value().count() << ";"
+                           << (applied ? "1" : "0") << std::endl;
                 }
             } else {
-                pass_applied = function_pass->run_on_model(func);
+                OPENVINO_THROW("The output file for logging transformation statistics is closed. "
+                               "Recording of statistics is not possible.");
             }
         }
+    }
 
-        if (m_visualize) {
-            // visualizations and serializations will be named after the outermost function
-            const size_t num_digits_in_pass_index = 3;
-            std::string index_str = std::to_string(index);
-            index_str = std::string(num_digits_in_pass_index - index_str.length(), '0') + index_str;
-            auto base_filename = func->get_name() + std::string("_") + index_str + std::string("_") + pass->get_name();
-
-            if (m_visualize) {
-                auto file_ext = "svg";
-                pass::VisualizeTree vt(base_filename + std::string(".") + file_ext);
-                vt.run_on_model(func);
+    void visualize(const shared_ptr<ov::Model>& model, const std::string& pass_name) const {
+        static size_t viz_index = 0;
+        if (m_visualize.is_enabled()) {
+            const auto& filter = m_visualize.get_str();
+            if (m_visualize.is_bool() || (pass_name.find(filter) != std::string::npos)) {
+                const auto& file_name = gen_file_name(model->get_name(), pass_name, viz_index++);
+                ov::pass::VisualizeTree vt(file_name + ".svg");
+                vt.run_on_model(model);
             }
         }
-        index++;
-        pass_timer.stop();
-        if (profile_enabled) {
-            cout << setw(7) << pass_timer.get_milliseconds() << "ms" << (pass_applied ? " + " : "   ")
-                 << pass->get_name() << "\n";
+    }
+
+    void serialize(const shared_ptr<ov::Model>& model, const std::string& pass_name) const {
+        static size_t serialize_index = 0;
+        if (m_serialize.is_enabled()) {
+            const auto& filter = m_serialize.get_str();
+            if (m_serialize.is_bool() || (pass_name.find(filter) != std::string::npos)) {
+                const auto& file_name = gen_file_name(model->get_name(), pass_name, serialize_index++);
+                ov::pass::Serialize serialize(file_name + ".xml", file_name + ".bin");
+                serialize.run_on_model(model);
+            }
         }
-        function_changed = function_changed || pass_applied;
-        needs_validate = pass_applied;
     }
-    if (profile_enabled) {
-        cout << "passes done in " << overall_timer.get_milliseconds() << "ms\n";
+
+private:
+    static std::string gen_file_name(const std::string& model_name, const std::string& pass_name, const size_t idx) {
```

Comment:
Use OV `file_util.hpp` for path name generation? The path can be char or wchar depends on environment.

Reply:
`visualize` and `serialize` methods, which use 'gen_file_name' method, save "xml+bin" or "svg" to the current directory.
the name of the file is generated by using the predefined vars (mode name, pass_name, id), which are std::string
This is already existed behavior, not a new one.